text/build_process.md

```diff
@@ -0,0 +1,122 @@
+
+* Daily development tasks
+ * Fast compile time
+ * Easy automated testing
+ * Easy dependency management
+ * Easy upgrade to the latest rustc nightly
+ * Should spot breakages between crates (while we stabilize the API)
+ * Fast CI results
+* Fallbacks
+  * Should not depend too much on Github (crates.io as another package repository)
+  * Should support forks (e.g.: iron-fork)
+* Community participation
+  * Writing a new adapter should be easy
+  * Contribution flow should be unique
+* Initial set up effort
+
+## Solutions
+
+### Keep small repos as is
+
+#### Use git submodules
+
+Git submodules have the same effect as pinning the git revision with Cargo.
+
+#### Always point to master
+
+One immediate solution could be to always point our dependencies to the latest master. This has 2 major consequences:
+
+* If a bottom-level dependency lands on master, we might break the compilation of another crate. Hence time a bottom-level dependency is updated, we have to run the tests of each crates depending on it. As a consequence, the bottom-level dependency has to know what actors are using it.
+
+* Rust ups will still have to be meticulously scheduled.
+
+#### Use Semantic Versioning
+
+Cargo [supports SemVer](http://doc.crates.io/crates-io.html#using-cratesio-based-crates). This less permissive approach would allow us to be explicit on when our API changes. For example, in Cargo.toml:
+``` toml
+taxonomy = "^0.4"
+```
+
+Based on an offline discussion with one Servo dev, it is worth to mention that Cargo might take 2 versions of the same crate. Consider this example:
+```toml
+taxonomy = "^0.4"
+```
+```toml
+taxonomy = "0.4.1"
+```
+If the most recent version of taxonomy is 0.4.2, Cargo will pick that one for the first build and pick 0.4.1 for the second one. Cargo won't pick the version that matches every constraint (which should have been 0.4.1). Therefore, we might build taxonomy twice, just like with master.
+
+Likewise, if somebody inadvertently breaks the API without increasing the major version (v0.5, in the example), we will break the other dependency. Thenceforth, we will need to run the tests of every actor using that dependency.
+
+#### Automate revisions bumps
+
+We could write a bumper script that will create a PR for each repo that depends on a bottom-level dependency and automatically lands it once all the PRs are green and an r+ was given on the initial PR. This seems to be the hardest solution in terms of initial set up. We will also rely more on Github than we currently do.
+
+### Have one repo with one single crate
+
+Bringing back taxonomy, thinkerbell and the adapters in foxbox will give a convenient way to find integration errors, as all the tests can be run locally without any particular inter-repository configuration.
+
+One major drawback would be the compile time. For more details about crate compile time, see [fxbox/foxbox#113](https://github.com/fxbox/foxbox/issues/113).
```

Comment:
Another drawback would be that it would considerably complicate experimenting. If I need to fix all the tree before I can start launching my first unit test, it's more complicated for me to test possible API changes.